src/Module.zig

```diff
@@ -4084,14 +4084,13 @@ pub fn namedFieldPtr(
             switch (child_type.zigTypeTag()) {
                 .ErrorSet => {
                     // TODO resolve inferred error sets
-                    const entry = if (val.castTag(.error_set)) |payload|
-                        (payload.data.fields.getEntry(field_name) orelse
-                            return mod.fail(scope, src, "no error named '{s}' in '{}'", .{ field_name, child_type })).*
+                    if (val.castTag(.error_set)) |payload|
+                        _ = (payload.data.fields.getEntry(field_name) orelse return mod.fail(scope, src, "no error named '{s}' in '{}'", .{ field_name, child_type }))
                     else
-                        try mod.getErrorValue(field_name);
+                        _ = try mod.getErrorValue(field_name);
 
                     const result_type = if (child_type.tag() == .anyerror)
-                        try Type.Tag.error_set_single.create(scope.arena(), entry.key)
+                        try Type.Tag.error_set_single.create(scope.arena(), field_name)
```

Comment:
The purpose of using `entry.key` here was that it is owned by the `Module` and has a longer lifespan than `field_name`.